### PR TITLE
Fix QTable add/insert row for multidimensional Quantity

### LINKED ISSUE #6072: Quantity insert() flattens by default for Table insert
As noted in a discussion on the AstroPy mailing list ("Vector Quantities in QTable"), there is a problem inserting a row into a Quantity if that column is multidimensional:
```
from astropy.table import QTable
import astropy.units as u
import numpy as np

foo = QTable(names=('state1','mass'),dtype=('(3,)f8','f8'))
foo['state1'].unit = u.K
foo['mass'].unit = u.mg

s0 = [235,240,245]*u.K
m0 = 5*u.g

foo.add_row((s0,m0))

ValueError: Unable to insert row because of exception in column 'state1':
Incorrect length for column state1 after inserting [ 235.  240.  245.] K (expected 3, got 1)
```
The problem stems from [newcol = col.insert(index, val)](https://github.com/astropy/astropy/blob/5e6c349679a42c1bed74ea37f7d95b2dc8a4580b/astropy/table/table.py#L2330) where the `insert` method is called with just `index` and `val`.  For `Column` there is an implied `axis=0`, but for `Quantity` the corresponding `insert` method has `axis=None` by default.  In that case the output array is flattened prior to insertion.  This matches the behavior of `ndarray` but doesn't work in the column context.

One solution is to change the `Column.insert()` API to add a kwarg `axis=0` and then in table call `insert(index, val, axis=0)`.  At this point we don't want to make the `Column` insert method flatten by default if axis is not supplied to avoid needless breakage.

@mhvk?

### PR DESCRIPTION
**Pull request by taldcroft**
Fixes #6072.

### CODE REVIEW

**Comment by pllim** [comment]
p.s. @bsipocz , that "under 100" goal was sweet while it lasted. 😉

**Comment by bsipocz** [comment]
@pllim - I'm sure we'll be down again after feature freeze.

File: astropy/table/column.py [hunk 1]
```diff
@@ -904,6 +904,9 @@ def insert(self, obj, values):
             Value(s) to insert.  If the type of ``values`` is different
             from that of quantity, ``values`` is converted to the matching type.
             ``values`` should be shaped so that it can be broadcast appropriately
+        axis : int, optional
+            Axis along which to insert ``values``.  If ``axis`` is None then
+            the column array is flattened before insertion.  Default is 0.
```
Comment (mhvk) [inline astropy/table/column.py]: Maybe extend the comment with
```
... Default is 0,
i.e., to insert new rows.
```

File: astropy/table/column.py [hunk 2]
```diff
@@ -1127,6 +1130,9 @@ def insert(self, obj, values, mask=None):
             ``values`` should be shaped so that it can be broadcast appropriately
         mask : boolean array_like
             Mask value(s) to insert.  If not supplied then False is used.
+        axis : int, optional
+            Axis along which to insert ``values``.  If ``axis`` is None then
+            the column array is flattened before insertion.  Default is 0.
```
Comment (mhvk) [inline astropy/table/column.py]: Same here, if we do it at all.

**Pull request merged**